caddy/shadowsocks_handler.go

```diff
@@ -120,15 +121,15 @@ func (h *ShadowsocksHandler) Handle(cx *layer4.Connection, _ layer4.Handler) err
 	switch conn := cx.Conn.(type) {
 	case transport.StreamConn:
 		h.service.HandleStream(cx.Context, conn)
-	case net.PacketConn:
+	case net.Conn:
 		n, err := cx.Read(h.buffer)
```

Comment:
Shouldn't we have a loop here to handle the association? Why handle only one packet?